crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -160,25 +60,41 @@ impl AuthorityStorePruner {
         Ok(pruned)
     }
 
-    fn process_checkpoints(
+    fn prune_objects(
         perpetual_db: &Arc<AuthorityPerpetualTables>,
         checkpoint_store: &Arc<CheckpointStore>,
         num_epochs_to_retain: u64,
     ) -> anyhow::Result<()> {
-        let mut pruned_seq_number = checkpoint_store
+        let mut checkpoint_number = checkpoint_store
             .get_highest_pruned_checkpoint_seq_number()?
             .unwrap_or_default();
+        let mut checkpoint_digest = CheckpointDigest::random();
         let current_epoch = checkpoint_store
             .get_highest_executed_checkpoint()?
             .map(|c| c.epoch())
             .unwrap_or_default();
+        let mut checkpoints_in_batch = 0;
+        let mut batch_effects = vec![];
+        let mut network_total_transactions = 0;
+
+        debug!(
+            "Starting object pruning. Current epoch: {}. Latest pruned checkpoint: {}",
+            current_epoch, checkpoint_number
+        );
         loop {
-            let Some(checkpoint) = checkpoint_store.get_checkpoint_by_sequence_number(pruned_seq_number + 1)?
-                else {return Ok(());};
+            let Some(checkpoint) = checkpoint_store.get_checkpoint_by_sequence_number(checkpoint_number + 1)? else {break;};
             // checkpoint's epoch is too new. Skipping for now
             if current_epoch < checkpoint.epoch() + num_epochs_to_retain {
```

Comment:
kind of. Code currently supports num_epochs_to_retain = 0, which implies aggressive pruner that tries to delete objects as soon as it can.
Code can be somewhat simplified if we agree not to support such behaviour and always keep at least one extra epoch of data